logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/UpdateClientVerificationStatusUseCase.kt

```diff
@@ -21,27 +21,37 @@ import com.wire.kalium.logic.StorageFailure
 import com.wire.kalium.logic.data.client.ClientRepository
 import com.wire.kalium.logic.data.conversation.ClientId
 import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.flatMap
 import com.wire.kalium.logic.functional.fold
 
 /**
- * Updates the verification status of a client.
+ * Updates the verification status of a client and user (if needed).
  * @param userId The user id of the client.
  * @param clientId The client id of the client.
  * @param verified The new verification status of the client.
  * @return [UpdateClientVerificationStatusUseCase.Result.Success] if the client was updated successfully.
  * [UpdateClientVerificationStatusUseCase.Result.Failure] if the client could not be updated.
  */
 class UpdateClientVerificationStatusUseCase internal constructor(
-    private val clientRepository: ClientRepository
+    private val clientRepository: ClientRepository,
+    private val userRepository: UserRepository
 ) {
     suspend operator fun invoke(userId: UserId, clientId: ClientId, verified: Boolean): Result =
-        clientRepository.updateClientVerificationStatus(userId, clientId, verified).fold(
-            { error -> Result.Failure(error) },
-            { Result.Success }
-        )
+        clientRepository.updateClientProteusVerificationStatus(userId, clientId, verified)
+            .flatMap {
+                clientRepository.getClientsByUserId(userId)
+                    .flatMap {
+                        val isUserVerified = it.all { client -> client.isProteusVerified }
+                        userRepository.updateProteusVerificationStatus(userId, isUserVerified)
+                    }
+            }
+            .fold({ error -> Result.Failure(error) },
+                { Result.Success }
```

Comment:
Do we want to log this info to Datadog? I guess it might not harm as it might be useful identifying and tracking possible issues in the future?